src/plugins/hosts/DrawerPanel/DrawerPanel.jsx

```diff
@@ -45,6 +53,40 @@ const DrawerPanel = forwardRef((props, ref) => {
     setOpen(false);
   };
 
+  /**
+   * Activate Plugin View
+   */
+  const activatePluginView = () => {
+    if (activeView === DRAWER.VIEWS.PLUGIN) toggleDrawer();
+    else openDrawer();
+    setActiveView(DRAWER.VIEWS.PLUGIN);
+  };
+
+  /**
+   * Activate Bookmark view
+   */
+  const activateBookmarkView = () => {
+    setActiveView(DRAWER.VIEWS.BOOKMARK);
+  };
+
+  /**
+   * Get active view
+   * @returns {string} Return current active view in drawer
+   */
+  const getActiveView = () => {
+    return activeView;
+  };
+
+  /**
+   * Reset state of active view
+   */
+  const resetDrawer = () => {
+    setActiveView(DRAWER.VIEWS.PLUGIN);
+    // Hack to always close the right drawer automatically
+    //  because we'll have tools that doesn't have any right menu
+    if (!initialOpenState) setOpen(initialOpenState);
```

Comment:
This was actually necessary to handle the following scenario:

> Where we were changing from an editor with the expanded right menu to a tool without any right menu.

In that case, if we don't have this line, the right menu stays open, but empty and without any way to close it.
So I added this `if` statement to keep at least the correct state of the left menu.

I suggest we move with this PR as it is already fixing a lot of issues with the `withBookmark` decorator and In the future, we can think of a better way to handle this.